remotehttp.go

```diff
@@ -34,13 +35,37 @@ func NewRemoteHTTPStore(location *url.URL, n int) (*RemoteHTTP, error) {
 		u.Path = u.Path + "/"
 	}
 
-	// Build a client with the right size connection pool and optionally disable
-	// certificate verification.
-	tr := &http.Transport{
-		DisableCompression:  true,
-		MaxIdleConnsPerHost: n,
-		TLSClientConfig:     &tls.Config{InsecureSkipVerify: TrustInsecure},
+	var tr *http.Transport
+
+	if cert != "" && key != "" {
+		// Load client cert
+		certificate, err := tls.LoadX509KeyPair(cert, key)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load client certificate from %s", cert)
+		}
+		caCertPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create CaCertPool")
+		}
+		tr = &http.Transport{
+			DisableCompression:  true,
+			MaxIdleConnsPerHost: n,
+			TLSClientConfig:     &tls.Config{
+				InsecureSkipVerify: TrustInsecure,
+				Certificates: []tls.Certificate{certificate},
+				RootCAs:      caCertPool,},
```

Comment:
If putting the `}` on the same line as the last entry, I'd argue that that defeats the value of the trailing comma (such being permitting new key/value pairs to be added without changing any other lines).